### PR TITLE
Add a Remove retired documentation section

### PR DESCRIPTION
**Pull request by sdake**
This is added to the about section. A flowchart is included to help contributors execute link reconciliation.

The source flowchart is in the team drive: https://docs.google.com/drawings/d/1w4q7ppw5HYFTEtV63Czs2DUl85Iwtju0fbZ7YvSUpuI/edit

Fixes: https://github.com/istio/istio.io/issues/7067

### CODE REVIEW

File: content/en/about/contribute/remove-content/index.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+To remove documentation from Istio, please follow these simple steps:
+
+1. Remove the page.
+1. Reconcile the broken links.
+1. Submit your contribution to GitHub.
+
+## Remove the page
+
+Use `git rm -rf` to remove the directory containing the page.
```
Comment (frankbu) [inline content/en/about/contribute/remove-content/index.md:16]:
```suggestion
Use `git rm -rf` to remove the directory containing the `index.md` page.
```

File: content/en/about/contribute/remove-content/index.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+## Reconcile broken links
+
+To reconcile broken links, use this flowchart:
+
+{{< image width="40%"
```
Comment (frankbu) [inline content/en/about/contribute/remove-content/index.md:22]: Enlarge diagram - to small to read in preview.

Comment (sdake) [reply content/en/about/contribute/remove-content/index.md:22]: set at 100% - will preview with our infra and resolve if it looks good.

File: content/en/about/contribute/remove-content/index.md (hunk 1 above)
Comment (sdake) [reply content/en/about/contribute/remove-content/index.md:16]: comment addressed.

File: content/en/about/contribute/remove-content/index.md (hunk 2 above)
Comment (sdake) [reply content/en/about/contribute/remove-content/index.md]: diagram looks to be correct size now (100%).

**Comment by frankbu** [comment]
One more small nit: the second YES line in the diagram is missing the arrowhead (and is also not perfectly horizontal).

Otherwise LGTM.

**Pull request merged**